integration/samples.sh

```diff
@@ -1,37 +1,60 @@
-#!/bin/bash
-set -e
-
-FIND_COMMAND=""
-
-if [[ $OSTYPE == "win32" ]]; then
-    FIND_COMMAND="dir /b /o:n /ad"
-else
-    FIND_COMMAND="find ./integration/samples/* -maxdepth 0 -type d"
-fi
-
-for D in $FIND_COMMAND
-do
-    P_JS=${D}/ng-package.js
-    P_JSON=${D}/ng-package.json
-    P_API=${D}/ng-packagr-api.js
-    P=${D}/package.json
-
-    if [ -f $P_API ]; then
-        echo "Building sample ${P_API}..."
-        node ${P_API}
-    elif [ -f $P_JS ]; then
-        echo "Building sample ${P_JS}..."
-        node dist/cli/main.js -p ${P_JS}
-    elif [ -f $P_JSON ]; then
-        echo "Building sample ${P_JSON}..."
-        node dist/cli/main.js -p ${P_JSON}
-    elif [ -f $P ]; then
-        echo "Building sample ${P}..."
-        if [ $P='failures' ]; then
-            node dist/cli/main.js -p ${P} || true
-        else
-            node dist/cli/main.js -p ${P}
+#!/bin/sh
+# NOTE: When making changes, make sure to run the script through ShellCheck!
+# @see https://github.com/koalaman/shellcheck
```

Comment:
I hope that it's OK to add this link here. This tool is very helpful in writing proper code. It reads the script bang and makes sure that only compatible syntax is used and also checks for a lot of features that might not work for the target shell.

Reply:
Yes it is